component/otelcol/auth/oauth2/oauth2_test.go

```diff
@@ -0,0 +1,155 @@
+package oauth2_test
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/grafana/agent/component/otelcol/auth"
+	"github.com/grafana/agent/component/otelcol/auth/oauth2"
+	"github.com/grafana/agent/pkg/flow/componenttest"
+	"github.com/grafana/agent/pkg/river"
+	"github.com/grafana/agent/pkg/util"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/config/configauth"
+	"gotest.tools/assert"
+)
+
+func TestConfigDefaults(t *testing.T) {
+	cfg := `
+		client_id     = "someclientid"
+		client_secret = "someclientsecret"
+		token_url     = "https://example.com/oauth2/default/v1/token"
+	`
+	var args oauth2.Arguments
+	require.NoError(t, river.Unmarshal([]byte(cfg), &args))
+
+	cfgWithExplicitDefaults := `
+		client_id       = "someclientid"
+		client_secret   = "someclientsecret"
+		token_url       = "https://example.com/oauth2/default/v1/token"
+		endpoint_params = {}
+		scopes          = []
+		timeout         = "0s"
+	`
+	var argsWithExplicitDefaults oauth2.Arguments
+	require.NoError(t, river.Unmarshal([]byte(cfgWithExplicitDefaults), &argsWithExplicitDefaults))
+
+	require.Equal(t, argsWithExplicitDefaults, args)
+}
+
+// Test performs a oauth2 integration test which runs the otelcol.auth.oauth2
+// component and ensures that it can be used for authentication.
+func Test(t *testing.T) {
+	tests := []struct {
+		testName      string
+		accessToken   string
+		tokenType     string
+		refreshToken  string
+		configBuilder func(string) string
+	}{
+		{
+			"runWithRequiredParams",
+			"TestAccessToken",
+			"TestTokenType",
+			"TestRefreshToken",
+			func(srvProvidingTokensURL string) string {
+				return fmt.Sprintf(`
+					client_id     = "someclientid"
+					client_secret = "someclientsecret"
+					token_url     = "%s/oauth2/default/v1/token"
+				`, srvProvidingTokensURL)
+			},
+		},
+		{
+			"runWithOptionalParams",
+			"TestAccessToken",
+			"TestTokenType",
+			"TestRefreshToken",
+			func(srvProvidingTokensURL string) string {
+				return fmt.Sprintf(`
+					client_id       = "someclientid2"
+					client_secret   = "someclientsecret2"
+					token_url       = "%s/oauth2/default/v1/token"
+					endpoint_params = {"audience" = ["someaudience"]}
+					scopes          = ["api.metrics"]
```

Comment:
Here I didn't add a test to check the `endpoint_params ` and `scopes `. I thought it might be an overkill, but I could add it if you think it's useful. Or maybe I can just write a todo in the code.

Reply:
I'll leave it as a todo for now